src/EditorFeatures/Core/Implementation/CommentSelection/ToggleLineCommentCommandHandler.cs

```diff
@@ -141,8 +157,40 @@ public bool ExecuteCommand(ToggleLineCommentCommandArgs args, CommandExecutionCo
                 }
             }
 
-            trackingSpans.Add(new CommentTrackingSpan(
-                TextSpan.FromBounds(linesInSelection.First().Start, linesInSelection.Last().End)));
+            var commentTrackingSpan = GetNewSelection(selectedSpan, editorOptions, isMultiCaret);
+            trackingSpans.Add(commentTrackingSpan);
+        }
+
+        private static CommentTrackingSpan GetNewSelection(SnapshotSpan selectedSpan, IEditorOptions editorOptions, bool isMultiCaret)
+        {
+            // If the span is just the caret location without a selection move the caret
+            // down to the next line at the same column (or end of line).
+            // We bail in multi caret scenarios as the next line could already be modified as part of this comment operation.
+            if (selectedSpan.IsEmpty && !isMultiCaret)
+            {
+                selectedSpan.End.GetLineAndCharacter(out var caretLineNumber, out var caretOffset);
+                var caretLine = selectedSpan.Snapshot.GetLineFromLineNumber(caretLineNumber);
+
+                var nextLineNumber = caretLineNumber + 1;
+                if (nextLineNumber < selectedSpan.Snapshot.LineCount)
+                {
+                    // Get the column in the line of the current caret location.
+                    var caretColumn = caretLine.GetColumnFromLineOffset(caretOffset, editorOptions);
+
+                    var nextLine = selectedSpan.Snapshot.GetLineFromLineNumber(nextLineNumber);
+                    // Compute the correct offset location from the column in the previous line.
```

Comment:
a super nit (you don't have to take this lol, I personally just think it looks a little cleaner with the newline):

```suggestion

                    // Compute the correct offset location from the column in the previous line.
```